File: AiStudio4/AiStudioClient/src/types/project.ts
// AiStudio4/AiStudioClient/src/types/project.ts

export interface Project {
  guid: string;
  name: string;
  path: string;
  description: string;
  createdDate: string;
  modifiedDate: string;
}

export interface ProjectFormValues {
  name: string;
  path: string;
  description: string;
}

export interface ProjectApiResponse {
  success: boolean;
  data?: Project | Project[];
  error?: string;
}

File: AiStudio4/Core/Models/Project.cs
// AiStudio4/Core/Models/Project.cs


namespace AiStudio4.Core.Models
{
    public class Project
    {
        public string Guid { get; set; } = System.Guid.NewGuid().ToString();
        public string Name { get; set; } = string.Empty;
        public string Path { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public DateTime CreatedDate { get; set; } = DateTime.UtcNow;
        public DateTime ModifiedDate { get; set; } = DateTime.UtcNow;
    }
}
